kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -439,294 +495,399 @@ define([
-            function selectAllOrNone(event) {
-                const selectAllChecked = event.target.checked;
+        /**
+         *
+         * @param {jquery element} $td the dom element with the file or folder icon
+         * @param {object} data the file data for that row
+         */
+        attachFileIconEvents: function ($td, data) {
+            if (data.isFolder) {
+                $td.find('button[data-name]')
+                    .off('click')
+                    .on('click', (e) => {
+                        $(e.currentTarget).off('click');
+                        const separator = this.path.slice(-1) !== '/' ? '/' : '';
+                        this.updatePathFn(this.path + separator + data.name);
+                    });
+                return;
+            }
 
-                //get all of the rows in the data table
-                const nodes = fullDataTable.fnGetNodes();
+            // First, we find the expansion caret in the first cell.
+            const $caret = $td.find('[data-caret]');
+
+            $caret.off('click');
+
+            // now, if there's openFileInfo on it, that means that the user had the detailed view open during a refresh.
+            if ($caret.length && this.openFileInfo[data.path]) {
+                //so we note that we've already loaded the info.
+                data.loaded = this.openFileInfo[data.path].loaded;
+                //toggle the caret
+                $caret.toggleClass('fa-caret-down fa-caret-right');
+                //and append the detailed view, which we do in a timeout in the next pass through to ensure that everything is properly here.
+                setTimeout(() => {
+                    $caret.parent().parent().after(this.renderMoreFileInfo(data));
+                }, 0);
+            }
 
-                $(`input.${cssBaseClass}-body__checkbox-input:enabled`, nodes)
-                    .prop('checked', selectAllChecked)
-                    .attr('aria-checked', selectAllChecked);
+            $caret.on('click', (e) => {
+                const fileExpander = $(e.currentTarget);
+                fileExpander.toggleClass('fa-caret-down fa-caret-right');
+                const $tr = fileExpander.parent().parent();
 
-                //enable or disable import appropriately
-                if (selectAllChecked) {
-                    stagingAreaViewer.enableImportButton();
+                if (fileExpander.hasClass('fa-caret-down')) {
+                    $('.kb-dropzone').css('min-height', '75px');
+                    this.openFileInfo[data.path] = data;
+                    $tr.after(this.renderMoreFileInfo(data));
                 } else {
-                    stagingAreaViewer.disableImportButton();
+                    $('.kb-dropzone').css('min-height', '200px');
+                    $tr.next().detach();
+                    delete this.openFileInfo[data.path];
                 }
+            });
+        },
+
+        updateTableDrawPath: function () {
```

Comment:
This uses some DataTables trickery to auto filter to the currently set path.
Based on this plug-in info: https://datatables.net/examples/plug-ins/range_filtering.html

Basically, this adds a search filter function that iterates over all rows. Any row that returns `true` passes the filter and gets shown.

This gets updated on redraw to use whatever the path is at the time.

Reply:
Very cool!

It seems as though there are numerous cases where embracing DataTables fully at the start would have saved time and trouble later on. I really like DT, it's powerful and very customisable.